### PR TITLE
feat(checkbox): indeterminate state

### PR DESCRIPTION
**Pull request by mJarsater**
**Describe pull-request**  
This PR introduces a new prop for the `tds-checkbox` - `indeterminate` state.

The prop displays the checkbox as indeterminate, it has a higher prio than checked. If the checkbox is `indeterminate` that will take precedence over `checked`.

Since `indeterminate` is a state of the checkbox, we can't know when it should be indeterminate or not, therefor that will be in the hands of the users.

This PR also introduces basic unit and e2e tests for the checkbox.

**Solving issue**  
Fixes: [CDEP-1589](https://tegel.atlassian.net/browse/CDEP-1589)

**How to test**  
1. Go to Checkbox
2. Use the controls to set the checkbox to a indeterminate state
3. Check the styling of the indeterminate state.
4. Check out the branch
5. Run `npm run test` in `packages/core`
6. Make sure all tests pass.

[CDEP-1589]: https://tegel.atlassian.net/browse/CDEP-1589?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by alexandereneroth** [comment]
Screenshot 2023-10-04 at 16 28 03
Hm I still get this issue, even after your fix

**Comment by alexandereneroth** [comment]
I suggest we simply treat `indeterminate` as a visual property that changes the appearance of a checked checkbox to avoid additional complexity.

This seems to be what ionicframework has done: https://ionicframework.com/docs/api/checkbox#indeterminate

**Comment by mJarsater** [comment]
> I suggest we simply treat `indeterminate` as a visual property that changes the appearance of a checked checkbox to avoid additional complexity.
> 
> 
> 
> This seems to be what ionicframework has done: https://ionicframework.com/docs/api/checkbox#indeterminate

Agree. I think that is pretty much what I have done. But I can remove the indeterminate key value from the events

**Comment by theJohnnyMe** [comment]
If I set the Intermediate control in Storybook to True and reload the page, Select All checkbox toggles between checked and intermediate state. The loop breaks once I start interacting with the other two checkboxes under it. 
I am not sure if it is desired behavior or not, so I wanted to bring it up to check.
https://github.com/scania-digital-design-system/tegel/assets/17004063/82688f9f-4f36-4913-8a9a-20edff3d34a2

**Comment by timrombergjakobsson** [comment]
> Screenshot 2023-10-04 at 16 28 03 Hm I still get this issue, even after your fix

How can this error not be consistent?

**Comment by timrombergjakobsson** [comment]
> > I suggest we simply treat `indeterminate` as a visual property that changes the appearance of a checked checkbox to avoid additional complexity.
> > This seems to be what ionicframework has done: https://ionicframework.com/docs/api/checkbox#indeterminate
> 
> Agree. I think that is pretty much what I have done. But I can remove the indeterminate key value from the events

Looks like that too me also, the implementation looks straightforward.

File: packages/core/src/components/checkbox/checkbox.tsx [hunk 1]
```diff
@@ -95,6 +109,9 @@ export class TdsCheckbox {
     return (
       <div class="tds-checkbox">
         <input
+          // eslint-disable-next-line no-return-assign
```
Comment (timrombergjakobsson) [inline packages/core/src/components/checkbox/checkbox.tsx:112]: I believe we have the same error in dropdown.tsx also.

**Comment by mJarsater** [comment]
> If I set the Intermediate control in Storybook to True and reload the page, Select All checkbox toggles between checked and intermediate state. The loop breaks once I start interacting with the other two checkboxes under it. I am not sure if it is desired behavior or not, so I wanted to bring it up to check.
> 
>  Screen.Recording.2023-10-05.at.07.42.19.mov

Could you try it out now? 

I will revert the story to only have one checkbox before we merge though, the ones in this PR is just for demo purposes.

File: packages/core/src/components/checkbox/test/checkbox.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { newE2EPage } from '@stencil/core/testing';
+
+describe('tds-checbox', () => {
```
Comment (alexandereneroth) [inline packages/core/src/components/checkbox/test/checkbox.e2e.ts:3]:
```suggestion
describe('tds-checkbox', () => {
```

**Review by theJohnnyMe — APPROVED** [review]
Nice work Martin!

**Pull request merged**